x-pack/plugins/alerting/server/alerts_client/alerts_client.test.ts

```diff
@@ -81,406 +82,310 @@ const mockSetContext = jest.fn();
 describe('Alerts Client', () => {
   let alertsClientParams: AlertsClientParams;
   let processAndLogAlertsOpts: ProcessAndLogAlertsOpts;
+
   beforeAll(() => {
     jest.useFakeTimers();
     jest.setSystemTime(new Date(date));
   });
 
-  beforeEach(() => {
-    jest.clearAllMocks();
-    logger = loggingSystemMock.createLogger();
-    alertsClientParams = {
-      logger,
-      elasticsearchClientPromise: Promise.resolve(clusterClient),
-      ruleType,
-      namespace: 'default',
-      rule: alertRuleData,
-      kibanaVersion: '8.9.0',
-    };
-    processAndLogAlertsOpts = {
-      eventLogger: alertingEventLogger,
-      ruleRunMetricsStore,
-      shouldLogAlerts: false,
-      flappingSettings: DEFAULT_FLAPPING_SETTINGS,
-      notifyWhen: RuleNotifyWhen.CHANGE,
-      maintenanceWindowIds: [],
-    };
-  });
-
   afterAll(() => {
     jest.useRealTimers();
   });
 
-  describe('initializeExecution()', () => {
-    test('should initialize LegacyAlertsClient', async () => {
-      mockLegacyAlertsClient.getTrackedAlerts.mockImplementation(() => ({
-        active: {},
-        recovered: {},
-      }));
-      const spy = jest
-        .spyOn(LegacyAlertsClientModule, 'LegacyAlertsClient')
-        .mockImplementation(() => mockLegacyAlertsClient);
-
-      const alertsClient = new AlertsClient(alertsClientParams);
-
-      const opts = {
-        maxAlerts,
-        ruleLabel: `test: rule-name`,
-        flappingSettings: DEFAULT_FLAPPING_SETTINGS,
-        activeAlertsFromState: {},
-        recoveredAlertsFromState: {},
-      };
-      await alertsClient.initializeExecution(opts);
-      expect(mockLegacyAlertsClient.initializeExecution).toHaveBeenCalledWith(opts);
-
-      // no alerts to query for
-      expect(clusterClient.search).not.toHaveBeenCalled();
-
-      spy.mockRestore();
-    });
-
-    test('should skip track alerts ruleType shouldWrite is false', async () => {
-      const spy = jest
-        .spyOn(LegacyAlertsClientModule, 'LegacyAlertsClient')
-        .mockImplementation(() => mockLegacyAlertsClient);
-
-      const alertsClient = new AlertsClient({
-        ...alertsClientParams,
-        ruleType: {
-          ...alertsClientParams.ruleType,
-          alerts: {
-            context: 'test',
-            mappings: { fieldMap: { field: { type: 'keyword', required: false } } },
-            shouldWrite: false,
-          },
-        },
-      });
-
-      const opts = {
-        maxAlerts,
-        ruleLabel: `test: rule-name`,
-        flappingSettings: DEFAULT_FLAPPING_SETTINGS,
-        activeAlertsFromState: {},
-        recoveredAlertsFromState: {},
-      };
-      await alertsClient.initializeExecution(opts);
-      expect(mockLegacyAlertsClient.initializeExecution).toHaveBeenCalledWith(opts);
-      expect(mockLegacyAlertsClient.getTrackedAlerts).not.toHaveBeenCalled();
-      spy.mockRestore();
-    });
+  for (const useDataStreamForAlerts of [false, true]) {
```

Comment:
A lot of the changes to jest tests are similar to this.  They now run the existing tests twice - once for not using data streams for alerts, once for using data streams as alerts.  You'll want "hide whitespace changes" on for these, though it seems like it's on by default for me now?  \o/